src/drawline.c

```diff
@@ -836,6 +838,10 @@ wlv_screen_line(win_T *wp, winlinevars_T *wlv, int negative_width)
 
 	for (int i = 0; i < 3 && i + skip < wp->w_width; ++i)
 	{
+	    if ((*mb_off2cells)(off, max_off) > 1)
+		// When the first half of a double-width character is
+		// overwritten, change the second half to a space.
+		ScreenLines[off + 1] = ' ';
```

Comment:
For 'listchars' "precedes", `MB_FILLER_CHAR` is used, but using it here leads to `>>><`, which looks a bit strange.

Reply:
Although it looks a bit strange, the computation of w_skipcol gets more complicated if we want to avoid it.  Also, using CTRL-E should not cause the text to shift left or right, thus we have this round-off for double width characters.
A nicer way would be to write the character on top of the last ">", thus we would only use two ">" characters, giving ">>##" (where "##" is the double width character).  But again this would be more code for covering a corner case.  We can leave it for later.